File: valo_tui/screens/about.py
"""[a] about — what valo-tui is, where the data comes from, and the keys."""

from __future__ import annotations

from textual.app import ComposeResult
from textual.containers import VerticalScroll
from textual.widgets import Label, Static

from .. import __version__
from .widgets import ACCENT, MUTED, TEXT


class AboutView(VerticalScroll):
    can_focus = True

    def compose(self) -> ComposeResult:
        yield Label("about", classes="page-title")
        body = (
            f"[bold {TEXT}]valo-tui[/]  [{MUTED}]v{__version__}[/]\n"
            f"[{TEXT}]A terminal-native tracker for global Valorant esports.[/]\n\n"
            f"[{ACCENT}]what you can do[/]\n"
            f"[{MUTED}]· browse events by region and stage[/]\n"
            f"[{MUTED}]· open an event for its overview, results, fixtures,[/]\n"
            f"[{MUTED}]  standings, bracket and teams[/]\n"
            f"[{MUTED}]· follow live matches with map scores across regions[/]\n"
            f"[{MUTED}]· drill into a match for veto, maps, agents, ACS,[/]\n"
            f"[{MUTED}]  K/D/A, ADR, HS%, FK and FD[/]\n\n"
            f"[{ACCENT}]data[/]\n"
            f"[{MUTED}]· source   [/][{TEXT}]vlr.gg (via vlrdevapi)[/]\n"
            f"[{MUTED}]· cache    [/][{TEXT}]SQLite, written by a background worker[/]\n"
            f"[{MUTED}]· the UI never blocks on the network[/]\n\n"
            f"[{ACCENT}]global nav[/]\n"
            f"[{MUTED}]· h [/][{TEXT}]home      [/][{MUTED}]landing dashboard[/]\n"
            f"[{MUTED}]· e [/][{TEXT}]events    [/][{MUTED}]pick a tournament to focus on[/]\n"
            f"[{MUTED}]· l [/][{TEXT}]live      [/][{MUTED}]global Bento dashboard[/]\n"
            f"[{MUTED}]· a [/][{TEXT}]about     [/][{MUTED}]this page[/]\n\n"
            f"[{ACCENT}]inside an event[/]\n"
            f"[{MUTED}]· o [/][{TEXT}]overview  [/][{MUTED}]· [/][{TEXT}]r [/][{MUTED}]results  [/]"
            f"[{TEXT}]· f [/][{MUTED}]fixtures[/]\n"
            f"[{MUTED}]· t [/][{TEXT}]standings [/][{MUTED}]· [/][{TEXT}]b [/][{MUTED}]bracket  [/]"
            f"[{TEXT}]· m [/][{MUTED}]teams[/]\n\n"
            f"[{ACCENT}]keys[/]\n"
            f"[{MUTED}]· ↑↓ / j k move · enter open · esc back to nav[/]\n"
            f"[{MUTED}]· ctrl+r refresh · q quit[/]\n"
        )
        yield Static(body, id="about-body")

    def load_data(self) -> None:  # for routing parity
        pass

File: internal/screens/about.go
package screens

import "strings"

// About is the [a] page: what valo-tui is, the data source, and the keys.
type About struct{ w, h int }

func NewAbout(w, h int) About { return About{w: w, h: h} }

func (s *About) SetSize(w, h int) { s.w, s.h = w, h }
func (s *About) Load()            {}

func (s About) View() string {
	var b strings.Builder
	b.WriteString(title("about") + "\n\n")
	b.WriteString(textB("valo-tui") + "  " + muted("v"+version) + "\n")
	b.WriteString(text("A terminal-native tracker for global Valorant esports.") + "\n\n")

	b.WriteString(accent("what you can do") + "\n")
	for _, l := range []string{
		"· browse events by region and stage",
		"· open an event for its overview, results, fixtures,",
		"  standings, bracket and teams",
		"· follow live matches with map scores across regions",
		"· drill into a match for veto, maps, agents, ACS,",
		"  K/D/A, ADR, HS%, FK and FD",
	} {
		b.WriteString(muted(l) + "\n")
	}
	b.WriteString("\n" + accent("data") + "\n")
	b.WriteString(muted("· source   ") + text("vlr.gg") + "\n")
	b.WriteString(muted("· cache    ") + text("SQLite, written by a background worker") + "\n")
	b.WriteString(muted("· the UI never blocks on the network") + "\n\n")

	b.WriteString(accent("global nav") + "\n")
	b.WriteString(muted("· h ") + text("home      ") + muted("landing dashboard") + "\n")
	b.WriteString(muted("· e ") + text("events    ") + muted("pick a tournament to focus on") + "\n")
	b.WriteString(muted("· l ") + text("live      ") + muted("global Bento dashboard") + "\n")
	b.WriteString(muted("· a ") + text("about     ") + muted("this page") + "\n\n")

	b.WriteString(accent("inside an event") + "\n")
	b.WriteString(muted("· o ") + text("overview  ") + muted("· ") + text("r ") + muted("results  ") + text("· f ") + muted("fixtures") + "\n")
	b.WriteString(muted("· t ") + text("standings ") + muted("· ") + text("b ") + muted("bracket  ") + text("· m ") + muted("teams") + "\n\n")

	b.WriteString(accent("keys") + "\n")
	b.WriteString(muted("· ↑↓ / j k move · enter open · esc back to nav") + "\n")
	b.WriteString(muted("· ctrl+r refresh · q quit"))
	return b.String()
}
